packages/qwik-city/adaptors/cloudflare-pages/vite/index.ts

```diff
@@ -40,7 +40,11 @@ export function cloudflarePagesAdaptor(opts: CloudflarePagesAdaptorOptions = {})
         const routesJson = {
           version: 1,
           include: [basePathname + '*'],
-          exclude: [basePathname + 'build/*', basePathname + 'assets/*'],
+          exclude: [
+            basePathname + 'build/*',
+            basePathname + 'assets/*',
+            basePathname + '*/q-data.json',
```

Comment:
This is not really the correct resolution here.
Adding these `q-data.json` paths to the `exclude` means that Pages will "never" make a request to the dynamic "Pages Function" to generate the data for a route.
This may work when using SSG if the `q-data.json` files are also generated alongside the `*.html` files, but that requires that the data is static and never changes.
In practice I would expect that the data might change over time and so statically generating all of it is not correct.

Instead of the approach here, each route should be able to set its own caching expiration for data requests via headers on the response. This way each endpoint can be cached for a suitable amount of time that appropriate for how volatile that data is.

Using response headers means that each layer of the network can cache as appropriate - whether that is Cloudflare's cache, an intermediate network cache, or the browser (or service worker).

Reply:
The symptom of this, in case anyone is experiencing it, is that the q-data.json requests are always 404 responses.

We had to workaround this problem by specifying a _routes.json file in our project manually that does not exclude these data requests.